client/my-sites/backup/backup-contents-page/file-browser/file-info-card.tsx

```diff
@@ -42,26 +49,63 @@ const FileInfoCard: FunctionComponent< FileInfoCardProps > = ( { siteId, item }
 	const [ isProcessingDownload, setIsProcessingDownload ] = useState< boolean >( false );
 	const downloadFile = useCallback( () => {
 		setIsProcessingDownload( true );
-		const manifestPath = window.btoa( item.manifestPath ?? '' );
-
-		wp.req
-			.get( {
-				path: `/sites/${ siteId }/rewind/backup/${ item.period }/file/${ manifestPath }/url`,
-				apiNamespace: 'wpcom/v2',
-			} )
-			.then( ( response: { url: string } ) => {
-				const downloadUrl = new URL( response.url );
-				downloadUrl.searchParams.append( 'disposition', 'attachment' );
-				window.open( downloadUrl, '_blank' );
-				setIsProcessingDownload( false );
-
-				dispatch(
-					recordTracksEvent( 'calypso_jetpack_backup_browser_download', {
-						file_type: item.type,
-					} )
-				);
-			} );
-	}, [ siteId, item, dispatch ] );
+
+		if ( item.type !== 'archive' ) {
+			const manifestPath = window.btoa( item.manifestPath ?? '' );
+			wp.req
+				.get( {
+					path: `/sites/${ siteId }/rewind/backup/${ item.period }/file/${ manifestPath }/url`,
+					apiNamespace: 'wpcom/v2',
+				} )
+				.then( ( response: { url: string } ) => {
+					const downloadUrl = new URL( response.url );
+					downloadUrl.searchParams.append( 'disposition', 'attachment' );
+					window.open( downloadUrl, '_blank' );
+					setIsProcessingDownload( false );
+
+					dispatch(
+						recordTracksEvent( 'calypso_jetpack_backup_browser_download', {
+							file_type: item.type,
+						} )
+					);
+				} );
+		} else {
+			if ( fileInfo === undefined || parentItem === undefined ) {
+				return;
+			}
+
+			let archiveType: string;
+			if ( fileInfo.dataType === 2 ) {
+				archiveType = 'plugin';
+			} else {
+				archiveType = 'theme';
+			}
+
+			const period = Math.round( rewindId );
```

Comment:
@jwebbdev I think it would be nice if the `path-info` or the `ls` API return the actual period for this extension. I rounded the `rewindId` that is coming from the URL as a workaround to test the functionality.

Reply:
The extensions don't actually have a reference period like other files.  They have manifests that they exist in (which would be what the URL is) and that's probably all we'd check.  The extensions themselves are stored separately when they're first encountered in any backup, and then all other backups (from any site) reference that global copy.